packages/flutter/lib/src/material/app.dart

```diff
@@ -64,11 +64,13 @@ class MaterialApp extends StatefulWidget {
     this.checkerboardOffscreenLayers: false,
     this.showSemanticsDebugger: false,
     this.debugShowCheckedModeBanner: true
-  }) : assert(debugShowMaterialGrid != null),
+  }) : this.initialRoute = initialRoute ?? ui.window.defaultRouteName ?? Navigator.defaultRouteName,
```

Comment:
I think we should rely on the runtime fallback of initialRoute in NavigatorState rather than duplicating it here.